src/modules/dirwatcher.js

```diff
@@ -0,0 +1,37 @@
+import EventEmitter from 'events'
+import fs from 'fs'
+import Set from 'collections/set'
+
+export default class DirWatcher extends EventEmitter {
+
+    constructor(path, delay) {
+        super();
+        this.allFiles = new Set();
+        this.watch(path, delay);
+    }
+
+    watch(path, delay) {
+        setInterval(() => {
+            fs.readdir(path, (error, files) => {
+                if (error) {
+                    console.log(error);
+                    throw error;
+                } else {
+                    if (this.allFiles.length == 0) {                        
+                        files.forEach(element => {
+                            this.allFiles.add(element);
+                        });                        
+                        this.emit("change", this.allFiles.map(filename => {return path + '/' + filename}));
+                    } else {                        
+                        const tempSet = new Set();
+                        files.forEach(element => {
+                            tempSet.add(element)
+                        });
+                        const diff = this.allFiles.difference(tempSet);                       
```

Comment:
As I mentioned before, please remove `collections` library. If you really want to use such methods like `difference` you can use `lodash` library, which has various methods for working with collections & objects. When you install it, for usage don't import the whole module but only the functions you need like
`import { difference } from 'lodash'`. Mainly all methods consume `Array/Set/Map`, but if Set&Map are not applicable, you can do next
```
const mySet = new Set(\*smth*\);
...
const setToArray = [...mySet] // ... is called spread/rest operator in JS
...
difference(setToArry, ...)
```

Docs here https://lodash.com/docs/4.17.11